src/applications/personalization/profile360/vet360/containers/ProfileField.jsx

```diff
@@ -31,68 +31,151 @@ import Vet360Transaction from '../components/base/Transaction';
 class Vet360ProfileField extends React.Component {
 
   static propTypes = {
-    clearErrors: PropTypes.func.isRequired,
     Content: PropTypes.func.isRequired,
     data: PropTypes.object,
     EditModal: PropTypes.func.isRequired,
     field: PropTypes.object,
     fieldName: PropTypes.string.isRequired,
     isEditing: PropTypes.bool.isRequired,
-    isEmpty: PropTypes.func,
-    onAdd: PropTypes.func.isRequired,
-    onCancel: PropTypes.func.isRequired,
-    onChange: PropTypes.func.isRequired,
-    onDelete: PropTypes.func.isRequired,
-    onEdit: PropTypes.func.isRequired,
-    onSubmit: PropTypes.func.isRequired,
-    refreshTransaction: PropTypes.func.isRequired,
+    isEmpty: PropTypes.bool.isRequired,
     title: PropTypes.string.isRequired,
     transaction: PropTypes.object,
     transactionRequest: PropTypes.object
   };
 
-  isEmpty = () => {
-    return this.props.isEmpty ? this.props.isEmpty(this.props) : !this.props.data;
+  onAdd = () => {
+    this.captureEvent('add-link');
+    this.openEditModal();
+  }
+
+  onCancel = () => {
+    this.captureEvent('cancel-button');
+    this.closeModal();
+  }
+
+  onChange = (value, property, skipValidation) => {
+    this.props.updateFormField(
+      this.props.fieldName,
+      this.props.convertNextValueToCleanData,
+      this.props.validateCleanData,
+      value,
+      property,
+      skipValidation
+    );
+  }
+
+  onDelete = () => {
+    let payload = this.props.data;
+    if (this.props.convertCleanDataToPayload) {
+      payload = this.props.convertCleanDataToPayload(payload, this.props.fieldName);
+    }
+
+    this.props.createTransaction(
+      this.props.apiRoute,
+      'DELETE',
+      this.props.fieldName,
+      payload,
+      this.props.analyticsSectionName
+    );
+  }
+
+  onEdit = () => {
+    this.captureEvent('edit-link');
+    this.openEditModal();
+  }
+
+  onSubmit = () => {
+    this.captureEvent('update-button');
+
+    let payload = this.props.field.value;
+    if (this.props.convertCleanDataToPayload) {
+      payload = this.props.convertCleanDataToPayload(payload, this.props.fieldName);
+    }
+
+    const method = payload.id ? 'PUT' : 'POST';
+
+    this.props.createTransaction(
+      this.props.apiRoute,
+      method,
+      this.props.fieldName,
+      payload,
+      this.props.analyticsSectionName
+    );
+  }
+
+  clearErrors = () => {
+    this.props.clearTransactionRequest(this.props.fieldName);
   }
 
-  isEditLinKVisible() {
+  closeModal = () => {
+    this.props.openModal(null);
+  }
+
+  openEditModal = () => {
+    this.props.openModal(this.props.fieldName);
+  }
+
+  refreshTransaction = () => {
+    this.props.refreshTransaction(this.props.transaction, this.props.analyticsSectionName);
+  }
+
+  captureEvent(actionName) {
+    recordEvent({
+      event: 'profile-navigation',
+      'profile-action': actionName,
+      'profile-section': this.props.analyticsSectionName,
+    });
+  }
+
```

Comment:
These were all moved from `mapDispatchToProps` so that we have access to the result of `mapStateToProps`.

Reply:
Previously we were doing an existence check on `actionName` in `captureEvent` but I don't see how that value could ever go undefined, so I simplified it by removing that check.